TGeant3/TGeant3TGeo.cxx

```diff
@@ -2179,21 +2179,30 @@ void TGeant3TGeo::SetSensitiveDetector(const TString &volumeName, TVirtualMCSens
       tgeoExtension = it->second;
    }
 
-   // Set extension to TGeoVolume with given name,
-   // print error message if volume does not exist or
-   // if the volume already has an extension
-   TGeoVolume *volume = gGeoManager->FindVolumeFast(volumeName);
-   if (!volume) {
-      Error("SetSensitiveDetector", "volume %s not found. Setting was ignored.\n", volumeName.Data());
-   } else {
-      // Issue a warning if the volume already has another extension
-      if (volume->GetUserExtension()) {
-         Warning("SetSensitiveDetector", "The existing extension of volume %s will be lost.\n", volumeName.Data());
-      }
-
-      // attach the extension to the volume
-      volume->SetUserExtension(tgeoExtension);
+   //
+   // Set extension to any instance of the TGeoVolume with the specified name.
+   // Extension holds the registered SD object.
+   //
+   //   Error if the named volume is not found.
+   //   Error if we overwrite a user's extension.
+   //
+   TIter next( gGeoManager->GetListOfVolumes() );
+   TGeoVolume* volume = 0;
+   while ( (volume = dynamic_cast<TGeoVolume*>( next() )) ) {
+     if (0==volume) {
+       Error("SetSensitiveDetector", "volume %s not found. Setting was ignored.\n", volumeName.Data());
+     }
```

Comment:
This Error message would not work es expected: it will not be issued if a volume of the given name does not exits. To detect such volume, you can add a counter before a loop, which will be incremented when a volume is found and tested after the loop.  The loop will be then:

```c++
   Int_t counter = 0;
   TIter next( gGeoManager->GetListOfVolumes() );
   TGeoVolume* volume = 0;
   while ( (volume = static_cast<TGeoVolume*>( next() )) ) {
      if ( volumeName != volume->GetName() ) continue;

      TGeoRCExtension* extension = static_cast<TGeoRCExtension*>( volume->GetUserExtension() );
      if ( extension && extension != tgeoExtension ) {
         Error("SetSensitiveDetector", "The existing extension of volume %s is overwritten by the SD implementation.\n", volumeName.Data());
      }
      volume->SetUserExtension(tgeoExtension);
      ++counter;
   }

   if (counter > 0) {
     if (gDebug > 0) {
       printf("SensitiveDetector set to %i volume(s) %s \n", counter, volumeName.Data());
     }
   }
   else {
     Error("SetSensitiveDetector", "Volume %s not found. Setting was ignored.\n", volumeName.Data());        
   }

```

Reply:
Guess I didn't have enough coffee when writing that.  Latest push should resolve.